torch/utils/data/datapipes/utils/common.py

```diff
@@ -20,6 +21,76 @@
 ]
 
 
+def validate_input_col(fn: Callable, input_col: Optional[Union[int, tuple, list]]):
+    """
+    Checks that function used in a map style datapipe works with the input column
```

Comment:
Let's list all the cases that are failing here to make sure users understanding the behavior of this function.